napari/_qt/qt_viewer.py

```diff
@@ -788,7 +799,14 @@ def on_resize(self, event):
         self.viewer._canvas_size = tuple(self.canvas.size[::-1])
 
     def _process_mouse_event(self, mouse_callbacks, event):
-        """Called whenever mouse pressed in canvas.
+        """Add properties to the mouse event before passing the event to the
+        napari events system. Called whenever mouse pressed in canvas.
+        This method adds following:
+            position: the position of the click in world coordinates
+            view_direction: a unit vector giving the direction of the camera in world coordinates
+            dims_displayed: a list of the dimensions currently being displayed in the viewer.
+            dims_point: the indices for the data in view
```

Comment:
```suggestion
        """Add properties to a mouse event before passing it on to callbacks.

        This method is called whenever a mouse is pressed within the napari
        canvas. It adds the following attributes to the mouse event:

        - ``position``: the position of the click in world coordinates.
        - ``view_direction``: a unit vector giving the direction of the click
          in world coordinates.
        - ``dims_displayed``: a list of the dimensions currently being
          displayed in the canvas.
        - ``dims_point``: the indices for the data in view.
```

and btw is dims_point in world or data coordinates?

Reply:
I see from the code that it's world. (I think!) Worth writing it in the docstring.

Reply:
Yes, it's world coordinates. I will add it to the docstring!